pennylane/fourier/spectrum.py

```diff
@@ -0,0 +1,272 @@
+# Copyright 2018-2021 Xanadu Quantum Technologies Inc.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Contains a transform that computes the frequency spectrum of a quantum
+circuit."""
+from itertools import product
+from functools import wraps
+import numpy as np
+import pennylane as qml
+
+
+def _get_spectrum(op):
+    r"""Extract the frequencies contributed by an input-encoding gate to the
+    overall Fourier representation of a quantum circuit.
+
+    If :math:`G` is the generator of the input-encoding gate :math:`\exp(-i x G)`,
```

Comment:
The spectrum we extract depends on G's eigenvalues for this definition. If you use the definition \exp(-i x/2 G) you'd have to divide the eigenvalues by half, and it is much easier to just subsume everything but i and x into the matrix.  So it is all consistent in the fourier module, and consistent with the papers. 

In the `Operation` class (non-user-facing) we define a generator as G in  \exp(-i a x G), and a are the coefficients. But this is due for a potential overhaul because a is not unique (you could just rescale the matrix), so I wouldn't base decisions on it - the way above is much cleaner!